lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -124,31 +110,64 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
         LOG.debug("The participant is < 18 years old and we do not have parental consent. Skipping record.")
         return None
 
-    patient_entry, patient_reference = create_patient(enrollment)
-    birthdate = parse_birth_date(enrollment)
+    # Create the participant resource entry and reference.
+    # Assumes that the project language is the participant's preferred language.
+    netid = normalize_net_id(enrollment.get('netid'))
+
+    if netid:
+        patient_entry, patient_reference = create_patient_using_unique_identifier(
+            sex = enrollment['core_sex'],
+            preferred_language = LANGUAGE_CODE[enrollment.project.id],
+            unique_identifier = netid,
+            record = enrollment,
+            system_identifier = INTERNAL_SYSTEM)
+    else:
+        patient_entry, patient_reference = create_patient_using_demographics(
+            sex = enrollment['core_sex'],
+            preferred_language = LANGUAGE_CODE[enrollment.project.id],
+            first_name = enrollment['core_participant_first_name'],
+            last_name = enrollment['core_participant_last_name'],
+            birth_date = enrollment['core_birthdate'],
+            zipcode = enrollment['core_zipcode'],
+            record = enrollment,
+            system_identifier = INTERNAL_SYSTEM)
 
     if not patient_entry:
         LOG.warning("Skipping record with insufficient information to construct patient")
         return None
 
-    location_resource_entries = locations(db, cache, enrollment)
+    birthdate = parse_date_from_string(enrollment.get('core_birthdate'))
+    if not birthdate:
+        LOG.warning("Record has an invalid or missing `core_birthdate` value")
+
+    location_resource_entries = build_residential_location_resources(
+        db = db,
+        cache = cache,
+        housing_type = enrollment.get('core_housing_type'),
+        primary_street_address = enrollment['core_home_street'],
+        secondary_street_address = enrollment['core_apartment_number'],
+        city = enrollment['core_home_city'],
+        state = enrollment['core_home_state'],
+        zipcode = enrollment['core_zipcode'],
+        system_identifier = INTERNAL_SYSTEM)
```

Comment:
Because the `build_residential_location_resources` function accepts information that comes from `core_*` fields, it would be better to given the function the `enrollment` object and allow it to parse for the various housing_type, primary_street_address, etc fields. 
Having fewer arguments in a function is generally considered better practice, making the function easier to maintain, call, and understand.

Reply:
In this refactor in redcap.py, I'm not having code access REDCap fields. Instead, the values are passed in as parameters, and accessing the REDCap fields is left to the caller. Once the "core" fields settle down (maybe it's me seeing them get used in another project) I'd be likely to add a build_residential_location_resources_using_core (or some name like that) type of functions to redcap.py.